shared-libs/cht-datasource/src/libs/error.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * Represents an error that occurs when an invalid argument is provided.
+ * This error is typically thrown when a function or method receives an argument
+ * that doesn't meet the expected criteria or constraints.
+ */
+export class InvalidArgumentError extends Error {
+  /**
+   * Constructor
+   * @param message a descriptive error message why the error was raised
+   */
+  constructor(message: string) {
+    super(message);
+    this.name = message;
+
+    Object.setPrototypeOf(this, InvalidArgumentError.prototype);
```

Comment:
This used to be how to do [prototypal inheritance](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Inheritance_and_the_prototype_chain#building_longer_inheritance_chains) before ES6 classes came along. Since we're using classes, this doesn't need to be set
```suggestion
```